lexers/troff.lua

```diff
@@ -7,24 +7,24 @@ local P, R, S = lpeg.P, lpeg.R, lpeg.S
 local lex = lexer.new(...)
 
 -- Registers and groff's structured programming.
-lex:add_rule('keywords', lex:tag(lexer.KEYWORD, (lexer.starts_line('.') * lexer.space^0 * (P('while') + P('break') + P('continue') + P('nr') + P('rr') + P('rnn') + P('aln') + P('\\}'))) + P('\\{'))
+lex:add_rule('keywords', lex:tag(lexer.KEYWORD, (lexer.starts_line('.') * lexer.space^0 * (P('while') + P('break') + P('continue') + P('nr') + P('rr') + P('rnn') + P('aln') + P('\\}'))) + P('\\{')))
 
 -- Markup.
 lex:add_rule('escape_sequences', lex:tag(lexer.VARIABLE,
-	P('\\') * ((P('s') * S('+-')) + S('*fgmnYV') + P("")) *
-	(P('(') * 2 + P('[') * (lexer.any-P(']'))^0 * P(']') + 1))
```

Comment:
The original man lexer appeared to use `S('BIPR')` instead of `1`. By removing `+ 1`, sequences like `\fB` no longer highlight. If troff shouldn't handle this sequence, then it should not replace man, but exist alongside it.

Please make sure you're testing your changes before submitting them for review. Otherwise this process is going to go a lot more slowly.

Reply:
`man` macros are troff syntax + special purpose macros. I have just generalized the highlighting from `man` specific macros to `troff` syntax (by adding strings, comments, more escape sequences, while retaining the macros specific to `man` (because man is extremely used and because those macros are shared by other macro sets, such as `ms`)). I did not intend to delete the +1 and do not really understand how I did, but the +1 is very important for the escape sequences to work correctly.